crates/bevy_render/src/render_phase/draw_state.rs

```diff
@@ -98,9 +94,14 @@ pub struct TrackedRenderPass<'a> {
 
 impl<'a> TrackedRenderPass<'a> {
     /// Tracks the supplied render pass.
-    pub fn new(pass: RenderPass<'a>) -> Self {
+    pub fn new(device: &RenderDevice, pass: RenderPass<'a>) -> Self {
+        let limits = device.limits();
         Self {
-            state: DrawState::default(),
+            state: DrawState {
```

Comment:
I may be missing something, but does this change means we always allocate a big buffer for every render pass? Is it small enough that it doesn't matter. The previous impl looked like it would just allocate memory when necessary, if you only need 2-3 bind groups this seems wasteful.

Reply:
The latter. The actual size of these Vecs aren't that big (1-2KB at most), and is a static upfront cost compared to the scaling cost of branching on every RenderCommand (not just every entity) that uses it.

Reply:
Alright, makes sense then, might be worth adding a small note in a comment I guess, but I'll approve anyway

Reply:
> The actual limits here are also usually quite small. 16 for both vertex buffers and bind groups, IIRC.

Some devices may report `u32::MAX` for those limits instead of an actual value, https://github.com/bevyengine/bevy/issues/6841 for example for `max_storage_buffers_per_shader_stage`. I don't know if that's also the case for `max_bind_groups` or `max_vertex_buffers` but that's a possibility

Reply:
After talking about this with cwfitzgerald about this on Discord, it seems like the limits for vertex buffers and bind groups are generally pretty low (max a few hundred). It seems like the limit is hard set in wgpu-hal mostly to avoid reallocations from Vecs.

In the worst case, we should set a reasonable upper limit ourselves (i.e. 1024) which we use `min` with the Device provided limits to be 100% sure it's not overallocating.

Original discussion: https://discord.com/channels/691052431525675048/743663924229963868/1059900749636702309

Reply:
Yeah, after reading the discussion I had the same conclusion.

Reply:
Ended up using `wgpu-hal`'s public constants `MAX_VERTEX_BUFFERS` and `MAX_BIND_GROUPS` so we're not keeping them out of sync.